fixed list function implementation (#1898)
pyccel/pyccel: CHANGELOG.md

-   #1853 : Fix translation of a file whose name conflicts with Fortran keywords.
-   Link and mention `devel` branch, not `master`.
-   #1047 : Print the value of an unrecognised constant.

> -   #1892 : Fix implementation of list function when list or tuple is passed as parameter.

Reviewer: I think this solution is more general than you suggest here. E.g. it should also work for sets. Maybe just mention iterables?
```suggestion
-   #1892 : Fix implementation of list function when an iterable is passed as parameter.
```
